### DISCUSSION TITLE
None unique UUIDs for workouts

### DISCUSSION THREAD

**Discussion by sandberg75** [kingstinct/react-native-healthkit#93] [Q&A]
Hello and thank you for this library.
I have been using it for some time now to sync health kit activities to my app.
I assumed that all activities had unique uuid's, but now I have noticed that I have several identical activities, (same sport, same start/end time etc) in the database, with different uuids.
37 users have duplicate activities registered due to none unique uuids, and 237 users are fine (no duplicate activities).
One of the users with duplicate activities is using a Garmin Watch to sync with AppleHealth, which in turn syncs with my app, could that be the reason of none unique uuids, have anyone experienced the same issue?

**Comment by robertherber** [comment]
Interesting. We haven't seen this issue - but in our current production apps we're still using the naive approach of only syncing events after the latest timestamp.

Could you ask the user with the Garmin to check if the data is duplicated already in Apple Health?

**Reply by sandberg75** [reply]
I did a small survey and it seems like events are not duplicated in Apple health, but it also confirmed that it is related to the Garmin users.

How do you handle delete/update of old activities if you only sync after latest timestamp, if I may ask?

**Reply by robertherber** [reply]
That would make me guess that the way Garmin syncs with HealthKit overwrites the previous entries in HealthKit instead of updating them - or what is your take on it? Do they provide some kind of "Garmin ID" in the metadata?

Currently we don't. It would be nice but we haven't had time to prioritise it.

**Reply by robertherber** [reply]
@sandberg75 In 8.1.1 it's now possible to query workouts with anchors :)
